### PR TITLE
feat(wallet): use new status-go collectibles backend

### LINKED ISSUE #11558: Background fetching of collectibles: Sync account collectibles with network every X minutes

### PR DESCRIPTION
**Pull request by dlipicar**
Part of https://github.com/status-im/status-desktop/issues/11379
Fixes https://github.com/status-im/status-desktop/issues/11558
Fixes https://github.com/status-im/status-desktop/issues/10467

status-go part: https://github.com/status-im/status-go/pull/3734

PR is split in two commits:
1) Implemented use of new backend
2) Removed old Collectibles service

Transaction Service / Module changes can be ignored, they will be removed in the near future.

For some reason the canFetchMore/fetchMore mechanism seems to be working worse than before. A workaround was introduced for now. To be investigated here: https://github.com/status-im/status-desktop/issues/11638

Collectibles are now fetched in the background by status-go.

This PR makes use of the new Async endpoints in status-desktop. A lot of pieces are missing (listed in the epic), but the following basic functionality is implemented here:
- [Collectibles are fetched in the background by status-go](https://github.com/status-im/status-desktop/issues/11558)
- Use new async endpoints
- [Filter by account and chainID lists](https://github.com/status-im/status-desktop/issues/10467)
- React to collectibles list change reported by status-go
- Transmit only collectible headers for the list (in chunks). Transmit full collectible details only when one is clicked. This should completely fix crashes due to large chunks of data being sent by the backend.
- React to new accounts added

https://github.com/status-im/status-desktop/assets/11161531/3518de1e-6e09-42a8-a5c4-dc2db866f0d3

### CODE REVIEW

**Review by caybro — CHANGES_REQUESTED** [review]
Massive work but...

pls check all the profile and showcase things still work as expected (I'm almost sure it will get broken with this PR)

**Comment by dlipicar** [comment]
@caybro agh sorry, raised this late yesterday and intended it to be a Draft PR, but messed up. This is still not ready for review. Will fix all that before publishing this.

File: ui/app/AppLayouts/Profile/views/profile/MyProfileSettingsView.qml [hunk 1]
```diff
@@ -211,7 +211,7 @@ ColumnLayout {
         ProfileShowcaseCollectiblesPanel {
             Layout.minimumHeight: implicitHeight
             Layout.maximumHeight: implicitHeight
-            baseModel: root.walletStore.flatCollectibles
+            baseModel: root.walletStore.collectibles
```
[the remark this answers is not in the document]
Comment (dlipicar) [reply ui/app/AppLayouts/Profile/views/profile/MyProfileSettingsView.qml:214]: There's actually one thing I need to discuss with you guys about `MyProfileSettingsView` and `ProfileShowcaseView`...
This is supposed to show all collectibles owned by some account (ours or some contact's) we're checking the profile of, right? The `flatCollectibles` model we were using before contained whatever we were showing in the collectibles tab, which is basically the assets owned by all/any derivated/watch-only accounts we've added in the wallet (from whatever chains we've selected). 
We probably need a dedicated controller to get the list of collectibles we're actually interested in for these two screens. For example, for a contact we want to check whatever address they're exposing, and for our own profile we probably don't want to include watch-only accounts, is that correct?

[the remark this answers is not in the document]
Comment (dlipicar) [reply ui/app/AppLayouts/Profile/views/profile/MyProfileSettingsView.qml:214]: The Wallet module gets its collectibles from the `Wallet/CollectiblesStore`
The Profile module gets its collectibles from the `Profile/WalletStore` instance. There, the collectibles model was renamed to `collectibles`. It points to the same one used by the wallet (which is wrong, the list of collectibles we show in the wallet and the one used for profile showcase are different, it needs to fetch them from the backend separately).

File: ui/imports/shared/stores/NetworkConnectionStore.qml [hunk 2]
```diff
@@ -13,7 +13,8 @@ QtObject {
 
     readonly property bool balanceCache: walletSectionAssets.hasBalanceCache
     readonly property bool marketValuesCache: walletSectionAssets.hasMarketValuesCache
-    readonly property bool collectiblesCache: walletSectionCollectibles.getHasCollectiblesCache()
+    //readonly property bool collectiblesCache: walletSectionCollectibles.getHasCollectiblesCache()
+    readonly property bool collectiblesCache: false
```
[the remark this answers is not in the document]
Comment (dlipicar) [reply ui/imports/shared/stores/NetworkConnectionStore.qml]: Added issue #

File: storybook/pages/ProfileDialogViewPage.qml [hunk 3]
```diff
@@ -325,7 +325,7 @@ SplitView {
                                 Component.onCompleted: append(data)
                             }
 
-                            readonly property var flatCollectibles: ListModel {
+                            readonly property var collectibles: ListModel {
```
[the remark this answers is not in the document]
Comment (dlipicar) [reply storybook/pages/ProfileDialogViewPage.qml:324]: See previous comment

File: ui/imports/shared/views/profile/ProfileShowcaseView.qml [hunk 4]
```diff
@@ -247,7 +247,7 @@ Control {
                 cellWidth: (width-rightMargin)/4
                 cellHeight: cellWidth
                 visible: count
-                model: root.isCurrentUser ? root.walletStore.flatCollectibles : null // TODO show other users too
+                model: root.isCurrentUser ? root.walletStore.ownedCollectibles : null // TODO get dedicated list of collectibles
```
[the remark this answers is not in the document]
Comment (dlipicar) [reply ui/imports/shared/views/profile/ProfileShowcaseView.qml]: See previous comment

**Comment by dlipicar** [comment]
@caybro cleanup done, please re-check

**Comment by caybro** [comment]
@dlipicar pls rebase; it doesn't even compile for me:
```
/home/ltinkl/git/status/status-desktop/src/backend/accounts.nim(389, 46) Error: type mismatch: got <string, string, string>
but expected one of:
proc loginWithKeycard(accountData: string; password: string; keyHex: string;
                      confNode: string): string
  first type mismatch at position: 4
  missing parameter: confNode

expression: status_go.loginWithKeycard($account, password, chatKey)
```

**Review by caybro — APPROVED** [review]
Code LGTM

But hard to test; I wasn't able to see any collectibles anywhere :)

image

**Comment by dlipicar** [comment]
> Code LGTM
> 
> But hard to test; I wasn't able to see any collectibles anywhere :)
> 

@caybro working fine for me 
image

I believe what you see is due to the fact that the current basic version will update collectibles when it's finished fetching them for all accounts. If you have many accounts with many collectibles, that can take a while. I'm working on the improvement to have the app be notified of partial fetches.

**Review by alaibe — APPROVED** [review]

**Review by stefandunca — APPROVED** [review]
Excellent job @dlipicar building a solid foundation for collectible access and moving business logic to status-go.

**Comment by anastasiyaig** [comment]
@dlipicar i was adding this acc as w/o and it crashed the app `0xd8d6ffe342210057bf4dcc31da28d006f253cef0`. seems to be fine in master but maybe i was just lucky? Does it sound known issue or indeed could be a regression in this PR? 

```
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/src/nim_status_client.nim(193) nim_status_client
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/src/nim_status_client.nim(190) mainProc
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimqml/src/nimqml/private/qguiapplication.nim(29) exec
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimqml/src/nimqml/private/qobject.nim(52) qobjectCallback
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimqml/src/nimqml/private/qobject.nim(37) onSlotCalled
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimbus-build-system/vendor/Nim/lib/core/macros.nim(561) onSlotCalled
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/src/app_service/service/wallet_account/service.nim(747) onAllTokensBuilt
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/src/app_service/service/wallet_account/dto.nim(216) getVisibleForNetworkWithPositiveBalance
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimbus-build-system/vendor/Nim/lib/pure/collections/tables.nim(244) []
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimbus-build-system/vendor/Nim/lib/system/assign.nim(179) genericSeqAssign
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimbus-build-system/vendor/Nim/lib/system/assign.nim(143) genericAssign
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimbus-build-system/vendor/Nim/lib/system/assign.nim(97) genericAssignAux
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimbus-build-system/vendor/Nim/lib/system/gc.nim(494) newSeq
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimbus-build-system/vendor/Nim/lib/system/gc.nim(486) newObj
/Users/jenkins/workspace/s_macos_aarch64_package_PR-11559/vendor/nimbus-build-system/vendor/Nim/lib/system/alloc.nim(735) rawAlloc
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

**Comment by dlipicar** [comment]
@anastasiyaig this seems to be the known bug when status-go sends a big chunk of data to nim, it crashes when deserializing (not always, depends on the app's memory usage in that instant). It's triggered by receiving the list of ERC20 tokens, not by anything related to collectibles, so I'd report a separate bug.

**Pull request merged**